app/frontend/src/search/utils.test.js

```diff
@@ -0,0 +1,50 @@
+import { constructNewUrlWithParam, stringMatchesPostcode, convertMilesToMetres } from './utils';
+
+describe('constructNewUrlWithParams', () => {
+    test('should activate autocomplete if threshold has been met', () => {
+        expect(constructNewUrlWithParam(
+            'location',
+            's',
+            'https://jobs?radius=20&location=&job_title=#vacancy-results'
+        )).toBe(
+            'https://jobs?radius=20&location=s&job_title=#vacancy-results'
+        );
+
+        expect(constructNewUrlWithParam(
+            'location',
+            'so',
+            'https://jobs?radius=20&location=s&job_title=#vacancy-results'
+        )).toBe(
+            'https://jobs?radius=20&location=so&job_title=#vacancy-results'
+        );
+    });
+});
+
+describe('stringMatchesPostcode', () => {
+    test('should test that string matching correct postcode is true', () => {
+        expect(stringMatchesPostcode('CT9 5ST')).toBe(true);
```

Comment:
My suggestion with these two tests would be to set up the correct and incorrect values in arrays within each test and then iterate them through the matcher.   That way it's clear that these tests are each only testing a single logical condition.